migrations/deploy.ts

```diff
@@ -0,0 +1,72 @@
+import { promises as fs } from 'fs';
+import { ethers, upgrades, network } from "hardhat";
+import { getAbi } from './tools/abi';
+import { Contract } from 'ethers';
+
+
+export function getContractKeyInAbiFile(contract: string) {
+    return contract.replace(/([a-zA-Z])(?=[A-Z])/g, '$1_').toLowerCase();
+}
+
+async function main() {
+
+    let production = false;
+    let upgradeable = false;
```

Comment:
It's safer to deploy upgradeable version by default.

Reply:
Agree